parsel/unified.py

```diff
@@ -66,26 +56,28 @@ class Selector(object_ref):
     }
     _lxml_smart_strings = False
 
-    def __init__(self, response=None, text=None, type=None, namespaces=None,
-                 _root=None, _expr=None):
-        self.type = st = _st(response, type or self._default_type)
+    def __init__(self, text=None, type=None, namespaces=None, _root=None, _expr=None):
+        self.type = st = _st(type or self._default_type)
         self._parser = _ctgroup[st]['_parser']
         self._csstranslator = _ctgroup[st]['_csstranslator']
         self._tostring_method = _ctgroup[st]['_tostring_method']
 
         if text is not None:
-            response = _response_from_text(text, st)
-
-        if response is not None:
-            _root = LxmlDocument(response, self._parser)
+            _root = self._get_root(text)
+        elif _root is None:
+            raise ValueError("Selector needs either text or _root argument")
 
-        self.response = response
         self.namespaces = dict(self._default_namespaces)
         if namespaces is not None:
             self.namespaces.update(namespaces)
         self._root = _root
         self._expr = _expr
 
+    def _get_root(self, text):
+        body = text.strip().encode('utf8') or '<html/>'
```

Comment:
if text is empty body is an unicode string, but if text is not empty body is a bytestring